### PR TITLE
Get stream url

### PR DESCRIPTION
**Pull request by rbuehlma**
This PR belongs to this discussion:
https://forum.kodi.tv/showthread.php?tid=318996&pid=2630102#pid2630102

It is intended as a replacement of strStreamURL

### CODE REVIEW

File: xbmc/addons/PVRClient.cpp [hunk 1]
```diff
@@ -1152,17 +1152,50 @@ bool CPVRClient::GetDescrambleInfo(PVR_DESCRAMBLE_INFO &descrambleinfo) const
   return false;
 }
 
-std::string CPVRClient::GetLiveStreamURL(const CPVRChannelPtr &channel)
+bool CPVRClient::FillLiveStreamFileItem(const CPVRChannelPtr &channel, CFileItem *fileItem)
```
Comment (FernetMenta) [inline xbmc/addons/PVRClient.cpp]: no need to pass channel anymore. it is a property of fileitem

File: xbmc/addons/PVRClient.cpp [hunk 2]
```diff
@@ -1152,17 +1152,50 @@ bool CPVRClient::GetDescrambleInfo(PVR_DESCRAMBLE_INFO &descrambleinfo) const
...
   PVR_CHANNEL tag;
   WriteClientChannelInfo(channel, tag);
-  strReturn = m_struct.toAddon.GetLiveStreamURL(tag);
-  return strReturn;
+  PVR_FILE_ITEM_PROPERTY* properties = new PVR_FILE_ITEM_PROPERTY[PVR_ADDON_MAX_FILE_ITEM_PROPERTIES];
+  int propertyCount = PVR_ADDON_MAX_FILE_ITEM_PROPERTIES;
+  char url[PVR_ADDON_URL_STRING_LENGTH];
+  int urlLength = PVR_ADDON_URL_STRING_LENGTH-1;
+  PVR_ERROR err = m_struct.toAddon.GetLiveStreamURL(tag, url, &urlLength, properties, &propertyCount);
+  if (err != PVR_ERROR_NO_ERROR) {
```
Comment (FernetMenta) [inline xbmc/addons/PVRClient.cpp]: wrong coding style

File: xbmc/pvr/addons/PVRClients.h [hunk 3]
```diff
@@ -282,11 +282,20 @@ namespace PVR
...
-     * @brief Get the stream URL for a given channel from the respective client.
-     * @param channel The channel to get the stream URL for.
-     * @return The stream URL or empty string if not available.
+     * @brief Fill the FileItem for a channel from the PVR backend.
+     * @param channel The channel to get the stream url for.
+     * @param fileItem The FileItem to be filled
+     * @return True if the url and and properties have been set
      */
-    std::string GetLiveStreamURL(const CPVRChannelPtr &channel);
+    bool FillLiveStreamFileItem(const CPVRChannelPtr &channel, CFileItem *fileItem);
```
Comment (FernetMenta) [inline xbmc/pvr/addons/PVRClients.h]: return value is never used

File: xbmc/addons/PVRClient.cpp (hunk 1 above)
Comment (rbuehlma) [inline xbmc/addons/PVRClient.cpp]: Removed the parameter, also for recordings.

File: xbmc/addons/PVRClient.cpp (hunk 2 above)
Comment (rbuehlma) [inline xbmc/addons/PVRClient.cpp]: Are you referring to the braces? I have corrected that.

File: xbmc/pvr/addons/PVRClients.h (hunk 3 above)
Comment (rbuehlma) [inline xbmc/pvr/addons/PVRClients.h]: I know, in the current case it does not matter as the following code should be executed in both cases. I still think its a good idea to have this return parameter available.

Would you prefer if I remove it?

**Comment by rbuehlma** [comment]
I added a commit. I will squash them all prior merging.

**Comment by ksooo** [comment]
@rbuehlma Many thanks. Your idea is conceptionally fine, but I have so many annotations and wishes for changes to the current code, that it is less time consuming if I come up with a PR superceeding yours.

**Pull request closed without merging**